dbms/src/Flash/tests/gtest_join_executor.cpp

```diff
@@ -75,213 +60,243 @@ class JoinExecutorTestRunner : public DB::tests::ExecutorTest
+        {toNullableVec<String>({"1", "1", {}, {}, "1", "1", {}}), toNullableVec<String>({{}, "3", {}, {}, {}, "3", {}}), toNullableVec<String>({"1", "1", "2", {}, "1", "1", {}}), toNullableVec<String>({"3", "3", "4", "3", {}, {}, {}})},
+        {toNullableVec<String>({{}, "1", "2", {}, "1", {}, {}}), toNullableVec<String>({"3", "3", "4", "3", "3", {}, {}}), toNullableVec<String>({"1", "1", "3", {}, {}, "1", {}}), toNullableVec<String>({"3", "3", "4", "3", "3", {}, {}})},
+        {toNullableVec<String>({{}, "1", "3", {}, "1", {}, {}}), toNullableVec<String>({"3", "3", "4", "3", "3", {}, {}}), toNullableVec<String>({"1", "1", "2", {}, {}, "1", {}}), toNullableVec<String>({"3", "3", "4", "3", "3", {}, {}})},
+        // semi join
+        {toNullableVec<String>({"1", "1"}), toNullableVec<String>({"3", {}})},
+        {toNullableVec<String>({"1", "1"}), toNullableVec<String>({"3", {}})},
+        {toNullableVec<String>({"1", "2", {}}), toNullableVec<String>({"3", "4", "3"})},
+        {toNullableVec<String>({"1", "3", {}}), toNullableVec<String>({"3", "4", "3"})},
+        // anti semi join
+        {toNullableVec<String>({"2", {}, {}}), toNullableVec<String>({"4", "3", {}})},
+        {toNullableVec<String>({"3", {}, {}}), toNullableVec<String>({"4", "3", {}})},
+        {toNullableVec<String>({"1", {}}), toNullableVec<String>({{}, {}})},
+        {toNullableVec<String>({"1", {}}), toNullableVec<String>({{}, {}})},
+        // left outer semi join
+        {toNullableVec<String>({"1", "2", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"1", "0", "0", "1", "0"})},
+        {toNullableVec<String>({"1", "3", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"1", "0", "0", "1", "0"})},
+        {toNullableVec<String>({"1", "2", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"1", "1", "1", "0", "0"})},
+        {toNullableVec<String>({"1", "3", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"1", "1", "1", "0", "0"})},
+        // anti left outer semi join
+        {toNullableVec<String>({"1", "2", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"0", "1", "1", "0", "1"})},
+        {toNullableVec<String>({"1", "3", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"0", "1", "1", "0", "1"})},
+        {toNullableVec<String>({"1", "2", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"0", "0", "0", "1", "1"})},
+        {toNullableVec<String>({"1", "3", {}, "1", {}}), toNullableVec<String>({"3", "4", "3", {}, {}}), toNullableVec<String>({"0", "0", "0", "1", "1"})},
```

Comment:
Why is the data type of match_helper_column not UInt8?
https://github.com/pingcap/tiflash/blob/245e2fbc586ff11df309d1f01d878d59a88c387b/dbms/src/Interpreters/Join.cpp#L112

Reply:
Same confusion. Let me look into it.

Reply:
Problem fixed.